test/unit/subcommands/test_deploy_subcommand.py

```diff
@@ -140,3 +147,66 @@ def rsa_key(*args, **kwargs):
         deploy_subcommand = DeploySubcommand()
         non_registered = deploy_subcommand._non_registered_slaves(registered_hosts, slaves_to_validate)
         self.assertEquals(0, len(non_registered))
+
+    @genty_dataset(
+        (
```

Comment:
What do you think about using the named datasets here (with keyword args to genty_dataset)? I feel like it definitely makes the test names more readable during the test run, and usually also makes it easier to comprehend what each set of test data is doing.